src/zproto_codec_c.gsl

```diff
@@ -2222,14 +2222,17 @@ $(class.name)_test (bool verbose)
         zlist_destroy (&$(message.name)_$(name));
 .       elsif type = "hash"
         zhash_t *$(name) = $(class.name)_get_$(name) (self);
+        // Order of values is not guaranted
 .           if defined (->test)
 .               for test
 .                   if first ()
-        assert (streq ((char *) zhash_first ($(field.name)), "$(test.)"));
-        assert (streq ((char *) zhash_cursor ($(field.name)), "$(test.name)"));
+        char *zhash_$(field.name)_value = (char *) zhash_lookup ($(field.name), "$(test.name)");
+        assert (zhash_$(field.name)_value != NULL);
+        assert (streq (zhash_$(field.name)_value, "$(test.)"));
 .                   else
-        assert (streq ((char *) zhash_next ($(field.name)), "$(test.)"));
-        assert (streq ((char *) zhash_cursor ($(field.name)), "$(test.name)"));
+        zhash_$(field.name)_value = (char *) zhash_lookup ($(field.name), "$(test.name)");
```

Comment:
@malanka don't we need a `char *` prefix here, too?

Reply:
On the line 2233 there is an explicit type conversion to char *. Where exactly do you think one more should be?

Reply:
To declare the variable. It is declared in 2229, but in the if-block, not in the else-block.

Maybe I am not familiar enough with gsl syntax, but I would expect in line 2233: 
`char *zhash_$(field.name)_value = (char *) zhash_lookup ($(field.name), "$(test.name)");`

Reply:
re-read gsl syntax and it's my fault, sorry